gobblin-compaction/src/main/java/org/apache/gobblin/compaction/verify/CompactionThresholdVerifier.java

```diff
@@ -60,7 +62,8 @@ public CompactionThresholdVerifier(State state) {
    * dataset. To avoid scalability issue, we choose a stateless approach where each dataset tracks
    * record count by themselves and persist it in the file system)
    *
-   * @return true iff the difference exceeds the threshold or this is the first time compaction
+   * @return true if the difference exceeds the threshold or this is the first time compaction or
```

Comment:
I am not sure if it is a good idea to package the logic of checking gmce-emission inside `thresholder` verifier.  What's the cost of having a separate implementation of verifier ?

Reply:
So the logic of verifier is if any of the verifier fail the dataset, the compaction will not run. In this case, if gmce verifier say it needs to re compact but threshold verifier say it does not need to be compacted, then the dataset will be skipped. That's the reason I embedded the logic here.